patroni/ha.py

```diff
@@ -766,8 +818,17 @@ def process_healthy_cluster(self):
                 if msg is not None:
                     return msg
 
-                return self.enforce_master_role('no action.  i am the leader with the lock',
-                                                'promoted self to leader because i had the session lock')
+                if self.cluster.is_standby_cluster():
+                    # in case of standby cluster we don't really need to
+                    # enforce anything, since the leader is not a master.
+                    # So just remind the role.
+                    self.enforce_follow_remote_master()
```

Comment:
same consideration: make enforce_follow_remote_master return you the lock you need.